packages/core/src/plugins/logStore/network/PropagationList.ts

```diff
@@ -0,0 +1,67 @@
+import { MessageRef } from '@streamr/protocol';
+
+export class PropagationList {
+	private readonly primaryItems: MessageRef[] = [];
+	private latestPrimary: MessageRef | undefined;
+	private isPrimaryFinalized: boolean = false;
+	private readonly foreignItems: MessageRef[] = [];
+	private isForeignFinalized: boolean = false;
+
+	public pushPrimary(messageRef: MessageRef) {
+		this.latestPrimary = messageRef;
+
+		const index = this.foreignItems.findIndex(
+			(item) => item.compareTo(messageRef) === 0
+		);
+
+		if (index != -1) {
```

Comment:
We need comments throughout this code too. I'll create an issue for this.

In essence, 
1. The Node is receiving a `QueryResponse` 
2. Pushing all items into this function
3. `foreignItems` is a map of messages that belong to this particular Foreign Node
4. In parallel, the `foreignItems` fills if the `primaryItems` (which is QueryResponse from all Nodes?) does not contain the message, and vice versa.
5. The result is returning the items that remain after this difference in items, if PN is final, or returning the range if not final.

Is this assessement correct?

Reply:
Perhaps, the source of the confusion is in the assumption that the `PropagationList` will get QueryResponses from all the Nodes.
That was the case which I already hae fixed.
Also I pushed test in `PropagationList.test.ts` that covers a lot of cases. Please take a look.